sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/linearRegression.scala

```diff
@@ -105,3 +106,100 @@ case class RegrAvgY(left: Expression, right: Expression)
       newLeft: Expression, newRight: Expression): RegrAvgY =
     this.copy(left = newLeft, right = newRight)
 }
+
+abstract class Regression
+  extends DeclarativeAggregate with ImplicitCastInputTypes with BinaryLike[Expression] {
+
+  override def nullable: Boolean = true
+  override def dataType: DataType = DoubleType
+  override def inputTypes: Seq[AbstractDataType] = Seq(DoubleType, DoubleType)
+
+  protected val count = AttributeReference("count", DoubleType)()
+  protected val meanX = AttributeReference("meanX", DoubleType)()
+  protected val meanY = AttributeReference("meanY", DoubleType)()
+  protected val c2 = AttributeReference("c2", DoubleType)()
+  protected val m2X = AttributeReference("m2X", DoubleType)()
```

Comment:
can you refer to some doc/wiki about the algorithm? otherwise it's hard to understand `c2` and `m2x`

Reply:
https://en.wikipedia.org/wiki/Algorithms_for_calculating_variance